libs/libapi/src/libapi/response.py

```diff
@@ -0,0 +1,63 @@
+import pyarrow as pa
+from datasets import Features
+from libcommon.s3_client import S3Client
+from libcommon.storage import StrPath
+from libcommon.storage_options import DirectoryStorageOptions, S3StorageOptions
+from libcommon.utils import MAX_NUM_ROWS_PER_PAGE, PaginatedResponse
+from libcommon.viewer_utils.features import to_features_list
+
+from libapi.utils import to_rows_list
+
+CACHED_ASSETS_S3_SUPPORTED_DATASETS: list[str] = ["asoria/image"]  # for testing
+
+
+def create_response(
+    dataset: str,
+    config: str,
+    split: str,
+    cached_assets_base_url: str,
+    cached_assets_directory: StrPath,
+    s3_client: S3Client,
+    cached_assets_s3_folder_name: str,
+    pa_table: pa.Table,
+    offset: int,
+    features: Features,
+    unsupported_columns: list[str],
+    num_rows_total: int,
+) -> PaginatedResponse:
+    if set(pa_table.column_names).intersection(set(unsupported_columns)):
+        raise RuntimeError(
+            "The pyarrow table contains unsupported columns. They should have been ignored in the row group reader."
+        )
+    storage_options = (
```

Comment:
Also here, you can directly use a S3StorageOptions

Reply:
Done.